### PR TITLE
Country National Society Projects Page

### PR DESCRIPTION
**Pull request by samshara**
## Changes
- Show national society project statistics
- Map view for the national society's projects
- List view of all the national society's ongoing projects

## TODO

-  Sankey Diagram

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

**Comment by samshara** [comment]
Let's merge this after https://github.com/IFRCGo/go-web-app/pull/83. 
For testing http://localhost:3000/countries/14/three-w/ns-projects is a good example.

File: src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+            type: 'Feature' as const,
+            geometry: {
+                type: 'LineString' as const,
+                coordinates: [
+                    countriesMap[relation.reportingNS]
+                        .centroid.coordinates as [number, number] ?? [0, 0],
+                    countriesMap[relation.projectCountry]
+                        .centroid.coordinates as [number, number] ?? [0, 0],
+                ],
+            },
```
Comment (tnagorra) [inline src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx]: We cannot use default [0, 0] for coordinates

File: src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+                id: country.id,
+                type: 'Feature' as const,
+                properties: {
+                    countryName: country.name,
+                    countryId: country.id,
+                    numProjects: countryProjects.length,
+                },
+                geometry: {
+                    type: 'Point' as const,
+                    coordinates: country.centroid?.coordinates as [number, number] ?? [],
```
Comment (tnagorra) [inline src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx]: We cannot use default [] for coordinates

File: src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+                countries,
+                projectList,
+                (project) => project.reporting_ns,
+            ),
+            projectsLineGeoJson: generateProjectsLineGeoJson(
+                countries,
+                projectList,
+            ),
+        } : {}),
+        [countries, projectList],
```
Comment (tnagorra) [inline src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx]: Let's just separate into different useMemo

File: src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+                                layerKey="arrow-icon"
+                                layerOptions={{
+                                    type: 'symbol',
+                                    paint: arrowPaint,
+                                    layout: iconReady ? arrowLayout : hiddenLayout,
+                                }}
+                            />
+                        </MapSource>
+                    )}
+                    {clickedPointProperties?.lngLat && clickedPointProperties.feature.id && (
```
Comment (tnagorra) [inline src/views/CountryThreeWNationalSocietyProjects/Map/index.tsx]: Let's use isDefined

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**